tests/butlerProxy.py

```diff
@@ -41,16 +41,17 @@
 class ButlerProxyTestCase(unittest.TestCase):
     """A test case for the data butler finding a Mapper in a root"""
 
-    def setUp(self):
-        if os.path.exists(os.path.join(ROOT, 'root/proxyOut')):
-            shutil.rmtree(os.path.join(ROOT, 'root/proxyOut'))
+    inputDir = os.path.join(ROOT, 'root')
+    outputDir = os.path.join(ROOT, 'ButlerProxyTestCase')
 
-        self.butler = dafPersist.Butler(os.path.join(ROOT, "root"), outPath="proxyOut")
+    def setUp(self):
+        self.tearDown()
+        self.butler = dafPersist.Butler(self.inputDir,
+                                        outPath=os.path.join(self.outputDir, "proxyOut"))
 
     def tearDown(self):
-        del self.butler
-        if os.path.exists(os.path.join(ROOT, 'root/out')):
-            shutil.rmtree(os.path.join(ROOT, 'root/out'))
+        if os.path.exists(self.outputDir):
+            shutil.rmtree(self.outputDir)
```

Comment:
It seems risky to have `setUp` call `tearDown`, since normally users are free to add things to `tearDown` without worrying about breaking other methods such as `setUp`. Instead, I suggest that you make a new method `deleteOutputDir` that does what `tearDown` does above, then have `setUp` and `tearDown` call that.

Note that it would then be safe to have `tearDown` delete the butler again, which demonstrates my point.